tests/Cms/Languages/LanguageConversionTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Data\Data;
+
+/**
+ * @coversDefaultClass \Kirby\Cms\Language
+ */
+class LanguageConversionTest extends TestCase
+{
+	public const TMP = KIRBY_TMP_DIR . '/Cms.LanguageConversion';
+
+	protected $app;
+
+	public function setUp(): void
+	{
+		Dir::make(static::TMP);
+	}
+
+	public function tearDown(): void
+	{
+		Dir::remove(static::TMP);
+	}
+
+	public function testConvertFromSingleLanguageToMultiLanguage(): void
+	{
+		$app = new App([
+			'roots' => [
+				'index' => static::TMP
+			],
+		]);
+
+		// create the page model
+		Data::write($app->root('content') . '/test/test.txt', [
+			'title' => 'Title'
+		]);
+
+		$this->assertCount(0, $app->languages());
+		$this->assertFalse($app->multilang());
+
+		// get the page model
+		$page = $app->page('test');
+
+		// and check if the content is loadable
+		$this->assertSame('Title', $page->content()->title()->value());
+
+		// create a new Language to switch to multi-language mode
+		Language::create([
+			'code' => 'en',
+		]);
+
+		// make sure that Kirby actually switched to multi-language mode
+		$this->assertCount(1, $app->languages());
+		$this->assertTrue($app->multilang());
+
+		// the content file should now have been moved to .en
+		$this->assertFileExists($page->root() . '/test.en.txt');
+		$this->assertFileDoesNotExist($page->root() . '/test.txt');
+
+		// the content should still be loadable for the page
+		$this->assertSame('Title', $page->content()->title()->value());
+		$this->assertSame('Title', $page->content('en')->title()->value());
```

Comment:
To verify that the existing content actually got moved correctly and isn't only cached in memory somewhere, I suggest we check the file contents as well.

Reply:
Good idea